### ISSUE TITLE
[BUG] Java references from JSP documents not matched

### ISSUE THREAD

**Issue by jmle** [konveyor/analyzer-lsp#389]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Konveyor version

0.3.0-beta.1.2

### Priority

Major

### Current Behavior

References to Java classes from within JSP documents are not matched.

For instance, this Java references is made from a JSP page:
![image](https://github.com/konveyor/analyzer-lsp/assets/4527008/dc85ca70-95ad-4ee7-9cc7-4f99e576b33f)
and matched in windup:
![image](https://github.com/konveyor/analyzer-lsp/assets/4527008/8543c75f-098a-439e-a25e-bc942966b7c4)
but not in the current analyzer.

### Expected Behavior

Java references within JSP documents should be matched, just like references made from other Java classes.

### How Reproducible

Always (Default)

### Steps To Reproduce

1. With the given environment, create a new rule somewhere in your local filesystem:
```yaml
- customVariables: []
  description: "Random description"
  message: "Random message"
  labels:
    - konveyor.io/source=eap6
    - konveyor.io/source=eap7
    - konveyor.io/source=eap
    - konveyor.io/target=eap8
  links: []
  ruleID: whatever-id-00001
  when:
    java.referenced:
      pattern: com.ibm.websphere.samples.daytrader.util.FinancialUtils
```
and save it with the filename you wish.
2. Use kantra to do an analysis:
```
./kantra analyze --target eap8 --input [...]sample.daytrader7 --mode full --output [...]sample.daytrader7.report --rules <your newly created rule>
```
3. Observe that no references to JSP files are made, only from Java files.

### Environment

```markdown
* Fedora 38
* Latest kantra version
* [sample.daytrader7 app (master branch)](https://github.com/WASdev/sample.daytrader7)
* Analysis configuration:
** Target: eap8
** Full mode
** Custom rule provided in "Steps to reproduce"
** No additional advanced options
```

**Comment by jmle** [comment]
@rromannissen FYI

**Comment by dymurray** [comment]
Closing due to inactivity or age, please reopen if you feel this was done in error
